Automattic/woocommerce-payments
# Fix deprecated string interpolation in Analytics.php

Fixes #7169 

#### Changes proposed in this Pull Request

**Title:** Fix deprecated string interpolation in Analytics.php

**Description:**
This PR addresses a deprecation issue in `/var/www/html/web/app/plugins/woocommerce-payments/includes/multi-currency/Analytics.php`. The problem involves the use of `${var}` for string interpolation, which is deprecated. This change replaces `${var}` with the correct `{$var}` format, resolving the following warnings:

- Deprecated: Using ${var} in strings is deprecated, use {$var} instead in /plugins/woocommerce-payments/includes/multi-currency/Analytics.php on line 330
- Deprecated: Using ${var} in strings is deprecated, use {$var} instead in /plugins/woocommerce-payments/includes/multi-currency/Analytics.php on line 331
- Deprecated: Using ${var} in strings is deprecated, use {$var} instead in /plugins/woocommerce-payments/includes/multi-currency/Analytics.php on line 332

**Testing instructions:**
To test this change, follow these steps:

No tests are required because the code changes involve straightforward syntax updates to resolve deprecation warnings. These changes are low-risk and don't introduce new functionality. Visual inspection and basic validation confirm the correctness of the fix. If further testing is required:

1. *Clone the repository and switch to the branch with this fix:*
   ```bash
   git clone https://github.com/your-username/your-repo.git
   cd your-repo
   git checkout bug-fix
2. *Ensure you have the necessary development environment set up for WooCommerce Payments.*
3. *Locate the Analytics.php file in the multi-currency folder within the WooCommerce Payments plugin.*
4. *Check lines 330-332 and the suggested fix*
5. *Identify test cases that previously triggered the deprecation warnings related to string interpolation using ${var}.*
6. *Execute the identified test cases and confirm that the deprecation warnings no longer appear.*
7. *Verify that the correct {$var} interpolation is now in place and that the affected code still functions as expected.*


-------------------

- [x] Run `npm run changelog` to add a changelog file (is `patch`).

**Post merge**
